pkg/controllers/node/tainting/controller.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tainting
+
+import (
+	"context"
+	"fmt"
+	golog "log"
+
+	"github.com/awslabs/operatorpkg/singleton"
+	"github.com/awslabs/operatorpkg/status"
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"sigs.k8s.io/karpenter/pkg/apis/v1beta1"
+
+	nodeutils "sigs.k8s.io/karpenter/pkg/utils/node"
+)
+
+var (
+	// conditionToTaintsMap maps condition values which can be present on the node to a set of taints that should be added in response
+	conditionToTaintsMap = map[string]v1.Taint{
```

Comment:
If every condition is added the same taint, what's the need of this map

Reply:
We probably don't **need** this, I wrote the initial implementation when we were still considering different taints. This does have the advantage of being easily expandable without adding significant complexity, but this could be reduced to just a set of status conditions in the short term.